benchmark-service/benchmark/benchmark.py

```diff
@@ -2,19 +2,64 @@
 import aiohttp_jinja2
 from aiohttp import web
 import logging
-from gear import setup_aiohttp_session, web_authenticated_developers_only
+from gear import setup_aiohttp_session, web_authenticated_developers_only, check_csrf_token
 from hailtop.config import get_deploy_config
 from hailtop.tls import get_in_cluster_server_ssl_context
 from hailtop.hail_logging import AccessLogger, configure_logging
-from web_common import setup_aiohttp_jinja2, setup_common_static_routes
-
+from web_common import setup_aiohttp_jinja2, setup_common_static_routes, render_template
+from benchmark.utils import ReadGoogleStorage, get_geometric_mean
+import json
+import re
+import plotly
+import plotly.express as px
 
 configure_logging()
 router = web.RouteTableDef()
 logging.basicConfig(level=logging.DEBUG)
 deploy_config = get_deploy_config()
 log = logging.getLogger('benchmark')
 
+FILE_PATH_REGEX = re.compile(r'gs://hail-benchmarks/((?P<user>[^/]+)/)((?P<version>[^-]+)-)((?P<sha>[^-]+))(-(?P<tag>[^\.]+))?\.json')
+default_filepath = 'gs://hail-benchmarks/tpoterba/0.2.45-ac6815ee857c-master.json'
+
+
+def get_benchmarks(file_path):
+
+    read_gs = ReadGoogleStorage()
+    file_info = read_gs.parse_file_path(file_path)
```

Comment:
Can you move the parsing of the file path to get the sha further down so we don't mix the reading logic with the parsing?